daemon/cmd/watchdogs.go

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"context"
+	"time"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/datapath/loader"
+	"github.com/cilium/cilium/pkg/endpoint"
+)
+
+const epBPFProgWatchdog = "ep-bpf-prog-watchdog"
+
+// checkEndpointBPFProgs check to ensure endpoints associated BPF programs remain loaded.
+func checkEndpointBPFProgs(d *Daemon, rate time.Duration) {
+	d.controllers.UpdateController(
+		epBPFProgWatchdog,
+		controller.ControllerParams{
+			Group: controller.NewGroup(epBPFProgWatchdog),
+			DoFunc: func(ctx context.Context) error {
+				var (
+					loaded = true
+					err    error
+					eps    = d.endpointManager.GetEndpoints()
+				)
+				for _, ep := range eps {
+					if ep.GetState() != endpoint.StateReady {
+						continue
+					}
+					loaded, err = loader.DeviceHasTCProgramLoaded(ep.HostInterface())
+					if err != nil {
+						log.WithField("ep_host_interface", ep.HostInterface()).
```

Comment:
We have well-defined logfields for these things already, see `pkg/logging/logfields` for the appropriate one.